rust/kernel/chrdev.rs

```diff
@@ -20,10 +20,66 @@ use crate::error::{Error, KernelResult};
 use crate::file_operations;
 use crate::types::CStr;
 
+struct SafeCdev(*mut bindings::cdev);
+
+impl SafeCdev {
+    fn alloc() -> KernelResult<Self> {
+        // SAFETY: call an unsafe function
```

Comment:
I 'm not sure if this is a final version meant for revision (in case it isn't, please ignore this), but we'd like to see a justification for the call being safe. Also, for consistency, please end the sentences in comments with a punctuation.

Reply:
I'm not sure that I understand the ins-and-outs of how we're planning to manage unsafe code in the kernel, so bear with me. How could I justify that this code is safe?